http/spec.go

```diff
@@ -68,16 +69,60 @@ func (s *Service) Spec(ctx context.Context,
 		return nil, err
 	}
 
-	data, metadata, err := decodeJSONResponse(bytes.NewReader(httpResponse.body), map[string]string{})
+	data, metadata, err := decodeJSONResponse(bytes.NewReader(httpResponse.body), map[string]any{})
 	if err != nil {
 		return nil, err
 	}
 
+	config := s.parseSpecsContainer(data)
+
+	// The application mask domain type is not provided by all nodes, so add it here if not present.
+	if _, exists := config["DOMAIN_APPLICATION_MASK"]; !exists {
+		config["DOMAIN_APPLICATION_MASK"] = phase0.DomainType{0x00, 0x00, 0x00, 0x01}
+	}
+	// The BLS to execution change domain type is not provided by all nodes, so add it here if not present.
+	if _, exists := config["DOMAIN_BLS_TO_EXECUTION_CHANGE"]; !exists {
+		config["DOMAIN_BLS_TO_EXECUTION_CHANGE"] = phase0.DomainType{0x0a, 0x00, 0x00, 0x00}
+	}
+	// The builder application domain type is not officially part of the spec, so add it here if not present.
+	if _, exists := config["DOMAIN_APPLICATION_BUILDER"]; !exists {
+		config["DOMAIN_APPLICATION_BUILDER"] = phase0.DomainType{0x00, 0x00, 0x00, 0x01}
+	}
+
+	s.spec = config
+
+	return &api.Response[map[string]any]{
+		Data:     s.spec,
+		Metadata: metadata,
+	}, nil
+}
+
+func (s *Service) parseSpecsContainer(data map[string]any) map[string]any {
 	config := make(map[string]any)
+
 	for k, v := range data {
+		// Handle complex structures
+		if k == "BLOB_SCHEDULE" {
+			if arrVal, isArr := v.([]map[string]any); isArr {
+				values := make([]map[string]any, len(arrVal))
+				for idx, val := range arrVal {
+					values[idx] = s.parseSpecsContainer(val)
+				}
```

Comment:
You should add a `continue` statement to stop processing of "BLOB_SCHEDULE" and move on to the next key.